shared/teams/team/invites/invite-history.tsx

```diff
@@ -0,0 +1,192 @@
+  const sections: Array<Section<Types.InviteLink>> = [
+    {
+      data,
+      key: 'invites',
+    },
+  ]
+
+  const emptyOrLoading =
+    loading || !data.length ? (
+      <Kb.Box2
+        direction="vertical"
+        fullWidth={true}
+        style={Styles.globalStyles.flexOne}
+        centerChildren={true}
+      >
+        {loading ? (
+          <Kb.Box2 direction="vertical" fullWidth={true} gap="tiny" centerChildren={true}>
+            <Kb.ProgressIndicator type="Large" />
+            <Kb.Text type="BodySmall">Loading...</Kb.Text>
+          </Kb.Box2>
+        ) : (
+          <Kb.Box2 direction="vertical" fullWidth={true} gap="tiny" centerChildren={true}>
+            <Kb.Text type="BodySmall">None yet.</Kb.Text>
+            <Kb.Button mode="Secondary" label="Generate invite link" onClick={onGenerate} />
+          </Kb.Box2>
+        )}
+      </Kb.Box2>
+    ) : null
+
+  const activeTitle = `Valid (${active.length})`
+  const expiredTitle = `Expired (${expired.length})`
+
+  return (
+    <Kb.Modal
+      header={{
+        hideBorder: true,
+        leftButton: Styles.isMobile ? (
+          <Kb.Text type="BodyBigLink" onClick={onClose}>
+            Close
+          </Kb.Text>
+        ) : (
+          undefined
+        ),
+        rightButton: Styles.isMobile ? (
+          undefined
+        ) : (
+          <Kb.Button mode="Secondary" label="Generate link" small={true} onClick={onGenerate} />
+        ),
+        title: <ModalTitle title="Invite links" teamID={teamID} />,
+      }}
+      footer={{
+        content: Styles.isMobile ? (
+          <Kb.Button fullWidth={true} mode="Secondary" label="Generate link" onClick={onGenerate} />
+        ) : (
+          <Kb.Button fullWidth={true} type="Dim" label="Close" onClick={onClose} />
+        ),
+        hideBorder: Styles.isMobile,
+      }}
+      onClose={onClose}
+      allowOverflow={true}
+      mode="DefaultFullHeight"
+      noScrollView={true}
+    >
+      <Kb.Tabs
+        tabs={[{title: activeTitle}, {title: expiredTitle}]}
+        onSelect={title => setShowingExpired(title === expiredTitle)}
+        selectedTab={showingExpired ? expiredTitle : activeTitle}
+        style={styles.tabs}
+      />
+      {emptyOrLoading ?? (
+        <Kb.BoxGrow>
+          <Kb.SectionList
+            sections={sections}
+            keyExtractor={item => item.id}
+            renderItem={({item}) => <InviteItem inviteLink={item} teamID={teamID} />}
+            contentContainerStyle={styles.listContent}
+            stickySectionHeadersEnabled={true}
+          />
+        </Kb.BoxGrow>
+      )}
+    </Kb.Modal>
+  )
+}
+
+const InviteItem = React.memo(
+  ({inviteLink, teamID}: {inviteLink: Types.InviteLink; teamID: Types.TeamID}) => {
+    const dispatch = Container.useDispatch()
+    const yourUsername = Container.useSelector(s => s.config.username)
+    const onExpire = () => dispatch(TeamsGen.createRemovePendingInvite({inviteID: inviteLink.id, teamID}))
```

Comment:
<img width="165" alt="image" src="https://user-images.githubusercontent.com/4698687/78720561-ef931000-78f3-11ea-84b1-64a8c9f0eab4.png">
onExpire should trigger this waiting state.

Reply:
a progress indicator might be nice too